tests/components/ambient_network/test_sensor.py

```diff
@@ -0,0 +1,163 @@
+from aioambient.errors import RequestError
+from freezegun import freeze_time
+import pytest
+
+from homeassistant.components import ambient_network
+from homeassistant.core import HomeAssistant
+
+from .conftest import setup_platform
+
+
+@freeze_time("2023-11-08")
+@pytest.mark.parametrize("config_entry", ["AA:AA:AA:AA:AA:AA"], indirect=True)
+async def test_sensors(
+    hass: HomeAssistant, open_api: OpenAPI, config_entry, aioambient
+) -> None:
+    """Test all sensors under normal operation."""
+    await setup_platform(hass, open_api, config_entry)
+
+    sensor = hass.states.get("sensor.station_a_absolute_pressure")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(977.61653)
+
+    sensor = hass.states.get("sensor.station_a_relative_pressure")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(1001.89694)
+
+    sensor = hass.states.get("sensor.station_a_daily_rain")
+    assert sensor is not None
+    assert sensor.state == "0.0"
+
+    sensor = hass.states.get("sensor.station_a_dew_point")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(27.77777)
+
+    sensor = hass.states.get("sensor.station_a_feels_like")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(29.44444)
+
+    sensor = hass.states.get("sensor.station_a_hourly_rain")
+    assert sensor is not None
+    assert sensor.state == "0.0"
+
+    sensor = hass.states.get("sensor.station_a_humidity")
+    assert sensor is not None
+    assert sensor.state == "60"
+
+    sensor = hass.states.get("sensor.station_a_last_rain")
+    assert sensor is not None
+    assert sensor.state == "2023-10-30T09:45:00+00:00"
+
+    sensor = hass.states.get("sensor.station_a_max_daily_gust")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(36.7252)
+
+    sensor = hass.states.get("sensor.station_a_monthly_rain")
+    assert sensor is not None
+    assert sensor.state == "0.0"
+
+    sensor = hass.states.get("sensor.station_a_irradiance")
+    assert sensor is not None
+    assert sensor.state == "37.64"
+
+    sensor = hass.states.get("sensor.station_a_temperature")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(28.27777)
+
+    sensor = hass.states.get("sensor.station_a_uv_index")
+    assert sensor is not None
+    assert sensor.state == "0"
+
+    sensor = hass.states.get("sensor.station_a_weekly_rain")
+    assert sensor is not None
+    assert sensor.state == "0.0"
+
+    sensor = hass.states.get("sensor.station_a_wind_direction")
+    assert sensor is not None
+    assert sensor.state == "11"
+
+    sensor = hass.states.get("sensor.station_a_wind_gust")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(14.75768)
+
+    sensor = hass.states.get("sensor.station_a_wind_speed")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(14.03347)
```

Comment:
Might as well look into snapshot testing. Checkout `analytics_insights` for an example you could drop in

Reply:
Added snapshot testing.